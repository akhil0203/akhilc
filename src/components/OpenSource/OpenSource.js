import React from 'react';

import './OpenSource.css';

const OpenSource = () => {
  return (
    <div className="OpenSource">
      <h1>OPEN SOURCE</h1>
    </div>
  );
};

export default OpenSource;
