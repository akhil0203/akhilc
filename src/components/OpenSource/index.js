import React from 'react';
import { Container, Col, Row } from 'reactstrap';
import { ParallaxBanner } from 'react-scroll-parallax';
import { Event } from '../Shared/Tracking';

import backgroundImg from '../../images/backgrounds/opensource.jpg';

import firefoxImg from '../../images/thumbnails/open-source/firefox.png';
import tensorflowImg from '../../images/thumbnails/open-source/tensorflow.png';
import railsImg from '../../images/thumbnails/open-source/rails.png';

import './styles.scss';

const OpenSource = () => {
  return (
    <ParallaxBanner
      className="open-source-parallax"
      layers={[
        {
          image: JSON.stringify(backgroundImg),
          amount: 0.25,
        },
      ]}
    >
      <Container className="open-source" fluid>
        <Col>
          <Row>
            <h2>Open Source</h2>
          </Row>
          <Col>
            <Row>
              <Col>
                <h3>40+ contributions across projects including:</h3>
                <p>
                  Rust, Homebrew, Rails, Tensorflow, AMO, Activity Stream and
                  more...
                </p>
              </Col>
            </Row>

            <Row>
              <Col sm={3}>
                <a
                  href="https://bit.ly/github_mozilla"
                  target="github-mozilla"
                  onClick={() =>
                    Event(
                      'Open Source',
                      "Clicked Mozilla PR's link",
                      "Mozilla PR's link",
                    )
                  }
                >
                  <img src={firefoxImg} alt="firefox" />
                </a>
              </Col>
              <Col sm={3}>
                <a
                  href="https://bit.ly/github_google"
                  target="github-tensorflow"
                  onClick={() =>
                    Event(
                      'Open Source',
                      'Clicked Tensorflow PR link',
                      'Tensorflow PR link',
                    )
                  }
                >
                  <img src={tensorflowImg} alt="tensorflow" />
                </a>
              </Col>
              <Col sm={3}>
                <a
                  href="https://bit.ly/github_rails"
                  target="github-rails"
                  onClick={() =>
                    Event(
                      'Open Source',
                      'Clicked Rails PR link',
                      'Rails PR link',
                    )
                  }
                >
                  <img src={railsImg} alt="rails" />
                </a>
              </Col>
            </Row>
          </Col>
        </Col>
      </Container>{' '}
    </ParallaxBanner>
  );
};

export default OpenSource;
