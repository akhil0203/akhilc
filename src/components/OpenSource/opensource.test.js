import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import OpenSource from '.';

it('matches the snapshot', () => {
  const tree = renderer.create(<OpenSource />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the OpenSource component without crashing', () => {
  shallow(<OpenSource />);
});

it('renders the main header', () => {
  const wrapper = shallow(<OpenSource />);

  expect(wrapper.find('h2').html()).toEqual('<h2>Open Source</h2>');
});

it('renders the subheader', () => {
  const wrapper = shallow(<OpenSource />);

  expect(wrapper.find('h3').html()).toEqual(
    '<h3>40+ contributions across projects including:</h3>',
  );

  expect(wrapper.find('p').html()).toEqual(
    '<p>Rust, Homebrew, Rails, Tensorflow, AMO, Activity Stream and more...</p>',
  );
});

it('renders an image for Firefox', () => {
  const wrapper = shallow(<OpenSource />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(1)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><a href="https://bit.ly/github_mozilla" target="github-mozilla"><img src="firefox.png" alt="firefox"/></a></div>',
  );
});

it('renders an image for Tensorflow', () => {
  const wrapper = shallow(<OpenSource />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(2)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><a href="https://bit.ly/github_google" target="github-tensorflow"><img src="tensorflow.png" alt="tensorflow"/></a></div>',
  );
});

it('renders an image for Rails', () => {
  const wrapper = shallow(<OpenSource />);

  expect(
    wrapper
      .find('Col>Row>Col')
      .at(3)
      .html(),
  ).toEqual(
    '<div class="col-sm-3"><a href="https://bit.ly/github_rails" target="github-rails"><img src="rails.png" alt="rails"/></a></div>',
  );
});
