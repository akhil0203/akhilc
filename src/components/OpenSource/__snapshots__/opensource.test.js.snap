// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="open-source container-fluid"
>
  <div
    className="col"
  >
    <div
      className="row"
    >
      <h2>
        Open Source
      </h2>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h3>
            40+ contributions across projects including:
          </h3>
          <p>
            Rust, Homebrew, Rails, Tensorflow, AMO, Activity Stream and more...
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <a
            href="https://bit.ly/github_mozilla"
            onClick={[Function]}
            target="github-mozilla"
          >
            <img
              alt="firefox"
              src="firefox.png"
            />
          </a>
        </div>
        <div
          className="col-sm-3"
        >
          <a
            href="https://bit.ly/github_google"
            onClick={[Function]}
            target="github-tensorflow"
          >
            <img
              alt="tensorflow"
              src="tensorflow.png"
            />
          </a>
        </div>
        <div
          className="col-sm-3"
        >
          <a
            href="https://bit.ly/github_rails"
            onClick={[Function]}
            target="github-rails"
          >
            <img
              alt="rails"
              src="rails.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
